### PR TITLE
Ping the server before updating job and step execution data

### PR DESCRIPTION
**Pull request by qrz-io**
During import and export jobs the MySQL server can become unavailable (because of timeouts, for example) and the connection will throw `The EntityManager is closed` and `MySQL Server has gone away` exceptions.

This is particularly troublesome for installations with Mongo, because the MySQL connection can be left untouched for large amounts of time, and then used to update Step and Job execution data. In this case, even though the product data was imported/exported successfully, the job will be marked as failed because the exception gets thrown at the moment of saving information about it.

To solve this, we can ensure our connection is alive by pinging the server and attempting to reconnect if it has become unavailable. 

Note: doctrine/dbal implements a [ping() method in 2.5](https://github.com/doctrine/dbal/blob/abbdfd1cff43a7b99d027af3be709bc8fc7d4769/lib/Doctrine/DBAL/Connection.php#L1571) but since Akeneo is using an older version, this PR adds a method to the doctrine job repository to do the same. Incidentally, in the comment for the ping method, doctrine recommends that it is used to ensure the server is available, like this PR is doing.

Note2: These PHP Warnings will still be issued if the server has indeed become unavailble:
```
PHP Warning:  PDO::query(): MySQL server has gone away in /var/www/html/akeneo/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php on line 801
PHP Warning:  PDO::query(): Error reading result set's header in /var/www/html/akeneo/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php on line 801
```

### CODE REVIEW

**Comment by loicguillois** [comment]
Hello,

Thanks a lot for contributing! Please take a moment to sign our contributor licence agreement : https://github.com/akeneo/pim-community-dev/blob/master/CONTRIBUTING.md. We will review your pull request once it's done.

Have a nice day,

**Comment by qrz-io** [comment]
Done :)

**Comment by juliensnz** [comment]
GTM

**Comment by damien-carcel** [comment]
GTM.

In Akeneo 1.6/master, we now use the 2.5 version of Doctrine/DBAL, so it will be possible to use the new `ping` method when merging this fix in 1.6.

File: src/Akeneo/Bundle/BatchBundle/Job/DoctrineJobRepository.php [hunk 1]
```diff
@@ -104,7 +106,40 @@ public function updateJobExecution(JobExecution $jobExecution)
...
+    }
+
+    /**
+     * Pings the server, returns false if it's not available.
+     * There is a ping() method in Doctrine\DBAL\Connection in the doctrine/dbal package
+     * as of 2.5.0, but  we are currently on 2.4.x
+     * @return bool
+     * @author Cristian Quiroz <cq@amp.co>
+     */
+    private function pingConnection()
```
Comment (solivier) [inline src/Akeneo/Bundle/BatchBundle/Job/DoctrineJobRepository.php]: Hi, we are not using private in our code but protected, could you change private to protected ?
I would add some warning log at least to say that we reconnected.

**Comment by solivier** [comment]
Hi @qrz-io,

Some comments has been added, could you aslo add some specs to test a bit the code ?
Once it's done we will be happy to merge your PR :)

Have a nice day.

**Comment by qrz-io** [comment]
Hi @solivier - apologies for the tardiness of this update, I've made the change you suggested (and removed an unnecessary parameter on a method call). I don't actually have a lot of experience with phpspec and haven't been able to find enough time to work on it (because of work and life) - so help on that would be much appreciated.

**Comment by nidup** [comment]
Hi @qrz-io !

Thank you for this contribution :tada: 

I cherry picked your commits in the following pull request https://github.com/akeneo/pim-community-dev/pull/5177 and I merged it to apply this improvement on the upcoming v1.7 (master branch) :tada: :tada: 

Why we don't merge it directly on v1.5? As we get more and more contributions, we continue to adapt our merging process. Merging on already released version requires lot of work for the core team, we have to merge on any upcoming versions (1.6 and 1.7 here), ensure everything is ok, resolve conflicts when the class has been changed since then (which is the case for job repository that has been improved in v1.6) and release every patches (v1.5.x, v1.6.x). We follow this process to ensure that we'll not cause any issue in production projects applying the last patch of a minor version and also ensure that if you update to a new minor version you well have the same fix and behavior applied. So we only merge most critical issues on already released versions and for any minor fixes or tech improvement we merge directly on master making them available in the upcoming minor version only.

Have a good day!

**Pull request closed without merging**